### ISSUE TITLE
ImageFFT.h missing from 1.4.0 package

### ISSUE THREAD

**Issue by o-smirnov** [casacore/casarest#8]
Getting this error building MeqTrees against the new casarest package:

```
/home/oms/meqtrees-timba/MeqNodes/src/UVBrick.cc:33:48: fatal error: casarest/synthesis/Images/ImageFFT.h: No such file or directory
 #include <casarest/synthesis/Images/ImageFFT.h>
```

File needs to be packaged, from the looks of it.

**Comment by gijzelaerr** [comment]
interesting, not sure what happened, it was working?

should be here somewhere I think

https://github.com/casacore/casarest/blob/master/synthesis/CMakeLists.txt#L8

**Comment by o-smirnov** [comment]
Well there's no install directive for ImageFFT.{cc,h}. That would explain it.

**Comment by gijzelaerr** [comment]
but I dont get it, i managed to get meqtrees compiling with casarest 1.3.x

2015-09-17 19:12 GMT+02:00 Oleg Smirnov <notifications@github.com>:

> Getting this error building MeqTrees against the new casarest package:
>
> ```/home/oms/meqtrees-timba/MeqNodes/src/UVBrick.cc:33:48: fatal error:
> casarest/synthesis/Images/ImageFFT.h: No such file or directory
> #include
>
>
> File needs to be packaged, from the looks of it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/casacore/casarest/issues/8>.
>



-- 
Gijs Molenaar
http://pythonic.nl

**Comment by o-smirnov** [comment]
This particular class used to be in casacore1, somehow got dropped in the move to casacore2, and moved to the casarest source tree instead. MeqTrees will compile against either of libcasacore{1,2}-dev. I bet you had libcasacore1-dev installed, and that allowed it to build. However it doesn't work right when built in that config (don't know how it even passed the travis tests!), hence my desire to get rid of casacore1 completely.

**Comment by gijzelaerr** [comment]
yeah but check the meqtrees package in the PPA, they should be compiled
against casacore 2 and casarest right

2015-09-17 19:21 GMT+02:00 Oleg Smirnov <notifications@github.com>:

> This particular class used to be in casacore1, somehow got dropped in the
> move to casacore2, and moved to the casarest source tree instead. MeqTrees
> will compile against either of libcasacore{1,2}-dev. I bet you had
> libcasacore1-dev installed, and that allowed it to build. However it
> doesn't work right when built in that config (don't know how it even passed
> the travis tests!), hence my desire to get rid of casacore1 completely.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/casacore/casarest/issues/8#issuecomment-141157812>.
>



-- 
Gijs Molenaar
http://pythonic.nl

**Comment by o-smirnov** [comment]
Fsck knows, I dunno how the package builds.... but I know it can't build without ImageFFT.h.

**Comment by gijzelaerr** [comment]
yeah i'm just worried that maybe something went wrong with the PR's and
stuff, but the history of the cmake file seems clean.

anyway, not a complex thing to do, but I think we also need to find out
what went wrong. Maybe I've been using the wrong repo or something, too
many repo's everywhere.

2015-09-17 19:26 GMT+02:00 Oleg Smirnov <notifications@github.com>:

> Fsck knows, I dunno how the package builds.... but I know it can't build
> without ImageFFT.h.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/casacore/casarest/issues/8#issuecomment-141158834>.
>



-- 
Gijs Molenaar
http://pythonic.nl

**Comment by o-smirnov** [comment]
Well, we clearly need an update to casarest with the files included, so could you please re-roll asap? I can't build on AWS without it.

**Comment by gijzelaerr** [comment]
@o-smirnov did you do a force push to the master?

I already fixed this problem and it was in master, but it is gone. I have no idea where it is now.

I'm *absolutely* sure I fixed the CMakefile to install these files:
```
casarest/synthesis/Images/ImageFFT.h .
casarest/synthesis/Images/ImageMetaData.h .
casarest/synthesis/Images/ImagePolarimetry.h .
```
And I applied other fixes also, but I did a git pull --rebase and now all my changes are gone.

**Comment by gijzelaerr** [comment]
Yeah here it is:

https://github.com/casacore/casarest/commit/f3949bc8ac42775cca9c2c1199c48fafeed65996

anyway, doing it again:

https://github.com/casacore/casarest/commit/c00584024de1b7dca1cad6b739a82efee278f258

will accept mac PR and make 1.4.1.

**Comment by o-smirnov** [comment]
Nope. Here's what I did according to my history:

 1575  git clone git@github.com:casacore/casarest.git
 1579  git checkout -b merge-lwimager-1.3.3-sep2015
 1595  git commit -a -m "merged in changes from ska-sa/lwimager"
 1596  git push

At this point it didn't let me push because I don't have write access to
the repo. So I forked the repo to o-smirnov/casarest, changed the remote of
my local copy, and pushed to that:

 1597  git status
 1598  git remote set-url origin git@github.com:o-smirnov/casarest.git
 1599  git remote -v
 1600  git push

I then issued a pull request from my fork.
